tools/posix/deploy_jenkins_slave.sh

```diff
@@ -100,10 +114,27 @@ download() {
     # Download one file and save its content to a given file name
     local url="$1"
     local output="$2"
-
-    echo ">>> Downloading $url"
-    echo "             to $output"
-    [ -f "$output" ] || curl --silent -L "$url" > "$output"
+    local try=1
+
+    # 5 tries, because we are generous
+    until [ ${try} -ge 6 ]; do
+        if [ -f "${output}" ]; then
+            if check_sum "${output}"; then
+                return
+            fi
+            rm -rf "${output}"
+        fi
+        echo ">>> [$try/5] Downloading $url"
+        echo "                   to $output"
+        set +e
+        curl --silent -L "$url" -o "$output"
+        set -e
```

Comment:
You should merge these 3 lines and avoid the explicit `set` statements (which don't belong in the middle of a script) using just `curl --silent -L "$url" -o "$output" || true`